File: src/main/java/com/medibox/MedicationAdherenceReminders/resource/ReminderResource.java
package com.medibox.MedicationAdherenceReminders.resource;

import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@With
public class ReminderResource {
  private Long id;

  @NotNull
  private Date createdDate;

  private Short pills;

  private Date endDate;

  @NotNull
  private MedicineResource medicine;

  @NotNull
  private Long userId;
  
  private FrequencyResource frequency;
  private IntervalResource interval;

  private Boolean consumeFood;
}

File: MedicationAdherenceMicroservice/Resources/ReminderResource.cs
namespace MedicationAdherenceMicroservice.Resources;

public class ReminderResource
{
    public long Id { get; set; }
    public DateTime CreatedDate { get; set; }
    public short? Pills { get; set; }
    public DateTime? EndDate { get; set; }
    public long UserId { get; set; }
    public MedicineResource Medicine { get; set; }
    public FrequencyResource? Frequency { get; set; }
    public IntervalResource? Interval { get; set; }
    public bool? ConsumeFood { get; set; }
}